### PR TITLE
chore: update Arrow dependencies

### PR DESCRIPTION
**Pull request by NGA-TRAN**
Closes #

This is my first PR and may miss many steps. I will sit down and talk with Edd today about this

- [ ] I've read the contributing section of the project [README](https://github.com/influxdata/influxdb_iox/blob/main/README.md).
- [ ] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed).

### CODE REVIEW

**Review by domodwyer — APPROVED** [review]
First PR! 🎉

File: query/src/frontend/sql.rs [hunk 1]
```diff
@@ -129,7 +129,7 @@ impl SQLQueryPlanner {
                 .context(CreatingTableProvider { table_name })?;
 
             ctx.inner_mut()
-                .register_table(&table_name, Box::new(provider));
+                .register_table(&table_name, Arc::new(provider));
```
Comment (NGA-TRAN) [inline query/src/frontend/sql.rs:132]: I am not sure why Rust compiler asked me to use Arc here. Did we change the way to use provider recently? Or maybe the way Arrow's Context work has changed? I may need to head to Arrow to read recent commits

Comment (domodwyer) [reply query/src/frontend/sql.rs:132]: It looks like @alamb changed it upstream in Arrow yesterday: https://github.com/apache/arrow/commit/29119e91c0e3efb65f4571fb00de15a60360aabd 👍

Comment (NGA-TRAN) [reply query/src/frontend/sql.rs:132]: Exactly what I am looking for. Thanks Dom

**Review by e-dard — APPROVED** [review]

**Review by alamb — COMMENTED** [review]
🎉

**Pull request merged**